.travis.yml

```diff
@@ -2,12 +2,33 @@ env:
+  - secure: gLBjXyAHpbFB8W6WLS6JOkM2mbp/ig01UMDqFWcJe0xx9t6eHiQ5RWA9Q3RAhv/9vfPMjYwmHHEKe+4D4uGuwK2iUOTHtXjJzX2qRIV9SHSIf/UNatFYpYeoZoYvu9Rx3/d6Ko1yW7ra9AMQd9nDSGXedJQjyP7SDEyJcVngu/2zAW3itVM93eMQbJJckyPBPU08HvyD3EbhXX6/HOxMvBe1/8cb1kmSgZVwmf9ljXlB4vNIXxrHRIIwlhYAbhYiNC/k6L4h6FvHSnRwHRLOohPqMZc3gOYZUDhIppYOBgt/IBL3IwLucWCFhJr5i9FsZ05XPagDznbpBfm0JV0D/dM1zMMF9Wt26lOb65+hn7xZTOrMj3OworY9iOuoOysSB0nV9WIf4oS7MjSbW+W+Q6J6ltD1ATTtRRucB96U76zBk+2mSPo+TV5EJmeEFjnt9/iFHRcKDaHEKDahAaGnx9+yNvyHv39EslqvLePSEm1GVZmYIE6IOkyIBRg7pTCesg0kPnF0wN/BzCkQgTmGF2+04JGG7yZCPb7tXXiaGDcThZLOGxiZmacI8wa8lAEpD1IcvLA6Hi0/pBeTMownVWLbUgmlsHuC/G0jhu++8w6Mkb08ig3hCrZGZeGM0vx5fdTRG2aQ7gck+wGJMCjKOnsjYJjqFq8Fui0YKInppAs=
+  - secure: d+Vix0H5X9QMHpu0Ip+tQTNt13itXpeLjmPO+Ur8xWbHBhGjztxS/fySnlGEla9i/II6NqB2Vd6Je5xwV/0VWKlE5mqmVBhiYSzRe0suo1DYkvOV9Jlm8pXXZ/NUcgnKNdkZT0WwLymdfFgt67r0mjxYKrPBC9nBpK89OLg2B61iHXeJK5W4F3pTdh5dwgaHXCcDGbKUan3a7r/60kOuc7DYdEphl8W6KAfF6hInT3JwTNXHE07OCF/xU+e9e4ox59n5YHBtURg8NCBbTwaHhuB59UfxAGekfaxTZxSlo6U0T1ha0gi4KtPxoP6FZQzTiAt/8BBeTuL2pv2702biK6mJVdtd2J0X/4KZTUnkSFJBNDshVda0ZpcgAHDw6oUIqYF0m5EZTcSMR0MWJOjgZL9U0Nas/XYSTtgf3N1Y7gtPKRoc2SgB3hV+XCOhQlE0DiZp87JT2vTMJc8gphgtp8jGIOPVF20U1O2gfcTJ7CUHAkJwb5TaMUNkObsfvlBXe1Wfh0Buzlx8Cs7TlnPrmn3xawb2T6r3+YPri9hz9xBkk8gZY5QSWknI6dfdOPQidw9UFa9tyoICsm4u2SK3fBEtnT94pogS0D1oaOjCl60cysmhNtkNvq7t1HmNaFy/WRBBQGW0CV5unRXrnoSCN3G2GtWa3fbd0W70eZiJ4N4=
 script:
+- export TRAVIS_TAG="pshutchinson/todo_app:${TRAVIS_COMMIT}"
 - docker build --target test --tag my-test-image .
 - docker run my-test-image todo_app/test_ViewModel.py
 - docker run my-test-image todo_app/test_app.py
-- docker run -e FLASK_APP -e FLASK_ENV -e SECRET_KEY -e TRELLO_KEY -e TRELLO_TOKEN -e TRELLO_BOARD_ID my-test-image todo_app/e2etests
\ No newline at end of file
+- docker run -e FLASK_APP -e FLASK_ENV -e SECRET_KEY -e TRELLO_KEY -e TRELLO_TOKEN
+  -e TRELLO_BOARD_ID my-test-image todo_app/e2etests
+- echo $DOCKER_PASSWORD | docker login --username "$DOCKER_USERNAME" --password-stdin
+- docker build --target production --tag "$TRAVIS_TAG" .
+- docker push "$TRAVIS_TAG"
+- docker build --target production --tag pshutchinson/todo_app:latest .
+- docker push pshutchinson/todo_app:latest
+- docker pull pshutchinson/todo_app:latest
+- echo $HEROKU_API_KEY | docker login --username=_ --password-stdin registry.heroku.com
+- docker tag pshutchinson/todo_app registry.heroku.com/pshutchinson-todoapp/web
+- docker push registry.heroku.com/pshutchinson-todoapp/web
+- heroku container:push web --app pshutchinson-todoapp
+- heroku container:release web --app pshutchinson-todoapp
```

Comment:
Note that you're pushing & deploying on any commit/PR to any branch currently; this would mean in a team with several developers a constant stream of updates, some of which are potentially acknowledged as broken. Instead we should make sure this is restricted to just named branches (`main` by default, but potentially temporarily individual branches like this one that want the deployment testing as well)
Probably the best way to do that would be to use the in-built [script deployment provider](https://docs.travis-ci.com/user/deployment/script/) - you could combine that with a `before_deploy` stage ([mentioned in the docs here](https://docs.travis-ci.com/user/job-lifecycle/#deploying-your-code)) if you think that makes it clearer.